packages/http-client-csharp/generator/Microsoft.Generator.CSharp.Input/src/InputTypes/InputEnumTypeIntegerValue.cs

```diff
@@ -1,9 +1,9 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
 namespace Microsoft.Generator.CSharp.Input
 {
-    internal class InputEnumTypeIntegerValue : InputEnumTypeValue
+    internal class InputEnumTypeIntegerValue : InputEnumTypeValue<int>
```

Comment:
Given that all of these abstractions which hold the value are classes and passed by reference into the serialization methods/converters, aren't they being boxed?

Reply:
their values are not, if we store it as object they will be.

Reply:
I don't think that's correct.   Value is a member of an object (InputEnumTypeIntegerValue) that lives on the heap and is passed by reference.    The Value member also lives on the heap as part of the instance, which boxes the int.    

To avoid boxing, this would need to be a `struct` and always passed by value so that it all flowed via stack